database/src/base.rs

```diff
@@ -0,0 +1,263 @@
+        account_id: &near_primitives::types::AccountId,
+    ) -> anyhow::Result<Vec<readnode_primitives::StateKey>>;
+
+    /// Returns state keys for the given account id filtered by the given prefix
+    async fn get_state_keys_by_prefix(
+        &self,
+        account_id: &near_primitives::types::AccountId,
+        prefix: &[u8],
+    ) -> anyhow::Result<Vec<readnode_primitives::StateKey>>;
+
+    /// Returns the state value for the given key of the given account at the given block height
+    async fn get_state_key_value(
+        &self,
+        account_id: &near_primitives::types::AccountId,
+        block_height: near_primitives::types::BlockHeight,
+        key_data: readnode_primitives::StateKey,
+    ) -> anyhow::Result<readnode_primitives::StateValue>;
+
+    /// Returns the near_primitives::account::Account at the given block height
+    async fn get_account(
+        &self,
+        account_id: &near_primitives::types::AccountId,
+        request_block_height: near_primitives::types::BlockHeight,
+    ) -> anyhow::Result<readnode_primitives::QueryData<near_primitives::account::Account>>;
+
+    /// Returns the contract code at the given block height
+    async fn get_contract_code(
+        &self,
+        account_id: &near_primitives::types::AccountId,
+        request_block_height: near_primitives::types::BlockHeight,
+    ) -> anyhow::Result<readnode_primitives::QueryData<Vec<u8>>>;
+
+    /// Returns the near_primitives::account::AccessKey at the given block height
+    async fn get_access_key(
+        &self,
+        account_id: &near_primitives::types::AccountId,
+        request_block_height: near_primitives::types::BlockHeight,
+        public_key: near_crypto::PublicKey,
+    ) -> anyhow::Result<readnode_primitives::QueryData<near_primitives::account::AccessKey>>;
+
+    #[cfg(feature = "account_access_keys")]
+    async fn get_account_access_keys(
+        &self,
+        account_id: &near_primitives::types::AccountId,
+        block_height: near_primitives::types::BlockHeight,
+    ) -> anyhow::Result<std::collections::HashMap<String, Vec<u8>>>;
+
+    /// Returns the near_primitives::views::ReceiptView at the given receipt_id
+    async fn get_receipt_by_id(
+        &self,
+        receipt_id: near_primitives::hash::CryptoHash,
+    ) -> anyhow::Result<readnode_primitives::ReceiptRecord>;
+
+    /// Returns the readnode_primitives::TransactionDetails at the given transaction hash
+    async fn get_transaction_by_hash(
+        &self,
+        transaction_hash: &str,
+    ) -> anyhow::Result<readnode_primitives::TransactionDetails>;
+
+    /// Returns the block height and shard id by the given block height
+    async fn get_block_by_height_and_shard_id(
+        &self,
+        block_height: near_primitives::types::BlockHeight,
+        shard_id: near_primitives::types::ShardId,
+    ) -> anyhow::Result<readnode_primitives::BlockHeightShardId>;
+}
+
+#[async_trait::async_trait]
+pub trait StateIndexerDbManager {
```

Comment:
Please, consider extracting these `StateIndexer...`, `Reader...` and `TxIndexer...` DbManager traits to separate files. Maybe, it's worth introducing a mod `base` similar to `scylladb` to be able to maintain it easier in the future?